runtime/major_gc.c

```diff
@@ -980,9 +980,11 @@ static void mark_slice_darken(struct mark_stack* stk, value child,
   }
 }
 
-CAMLno_tsan /* Loads from locations in the OCaml heap can cause false alarms in
-               TSan when these locations are concurrently written to by
-               caml_modify (see comment inside the function). */
+static Caml_noinline CAMLno_tsan
```

Comment:
Consider making this `Caml_noinline` only `#if defined(WITH_THREAD_SANITIZER)` so as to not impact non-tsan builds.

Reply:
Good idea, thanks. cd94145